xray/src/backend.rs

```diff
@@ -11,20 +10,31 @@ use std::io;
 use std::path::PathBuf;
 use std::sync::Arc;
 use urlencoded::UrlEncodedQuery;
+use Meta;
 
-#[derive(Serialize, Debug)]
-struct BoundingRect {
+#[derive(Clone, Serialize, Debug)]
+pub struct BoundingRect {
     min_x: f32,
     min_y: f32,
     edge_length: f32,
 }
 
 #[derive(Serialize, Debug)]
-struct NodeMeta {
+pub struct NodeMeta {
     id: String,
     bounding_rect: BoundingRect,
 }
 
+impl NodeMeta {
+    pub fn id(&self) -> String {
+        self.id.clone()
+    }
+
+    pub fn bounding_rect(&self) -> BoundingRect {
+        self.bounding_rect.clone()
+    }
+}
```

Comment:
do you need these two getter functions? Looks like they are never called.

Reply:
I do need them, but I will instead just make the fields public.